grow/preprocessors/google_drive.py

```diff
@@ -161,74 +164,140 @@ def _update_node(root, part):
-        content = GoogleSheetsPreprocessor.download(
-            path=path, sheet_id=sheet_id, gid=gid, logger=self.pod.logger)
-        existing_data = None
-        if (path.endswith(('.yaml', '.yml'))
-                and self.config.preserve
-                and self.pod.file_exists(path)):
-            existing_data = self.pod.read_yaml(path)
-        content = GoogleSheetsPreprocessor.format_content(
-            content=content, path=path, format_as=self.config.format,
-            preserve=self.config.preserve,
-            existing_data=existing_data, key_to_update=key_to_update)
-        content = GoogleSheetsPreprocessor.serialize_content(
-            formatted_data=content, path=path,
-            output_style=self.config.output_style)
-        self.pod.write_file(path, content)
-        self.logger.info('Downloaded Google Sheet -> {}'.format(path))
+        spreadsheet_id = config.id
+        gids = config.gids or []
+        if config.gid is not None:
+            gids.append(config.gid)
+        format_as = config.format
+        if config.collection and format_as not in ['map', 'string']:
+            format_as = 'map'
+        gid_to_sheet, gid_to_data = GoogleSheetsPreprocessor.download(
+            spreadsheet_id=spreadsheet_id, gids=gids, format_as=format_as,
+            logger=self.pod.logger)
+
+        if config.path:
+            # Single sheet import.
+            path, key_to_update = self._parse_path(config.path)
+
+            for gid in gids:
+                # Preserve existing yaml data.
+                if (path.endswith(('.yaml', '.yml'))
+                        and self.config.preserve and self.pod.file_exists(path)):
+                    existing_data = self.pod.read_yaml(path)
+                    gid_to_data[gid] = utils.format_existing_data(
+                        old_data=existing_data, new_data=gid_to_data[gid],
+                        preserve=self.config.preserve, key_to_update=key_to_update)
+
+                content = GoogleSheetsPreprocessor.serialize_content(
+                    formatted_data=gid_to_data[gid], path=path,
+                    output_style=self.config.output_style)
+
+                self.pod.write_file(path, content)
+                self.logger.info('Downloaded Google Sheet -> {}'.format(path))
+        else:
+            # Multi sheet import.
+            collection_path = config.collection
+
+            if not gids:
+                gids = gid_to_sheet.keys()
+
+            for gid in gids:
+                if gid_to_sheet[gid]['title'].strip().startswith('_'):
+                    continue
+                file_name = '{}.yaml'.format(
+                    utils.slugify(gid_to_sheet[gid]['title']))
+                output_path = os.path.join(collection_path, file_name)
+                self.pod.write_yaml(output_path, gid_to_data[gid])
+                self.logger.info(
+                    'Downloaded {} -> {}'.format(gid_to_sheet[gid]['title'], output_path))
```

Comment:
Would be good to include the GID after the title, just in case the user wants to go and update the `gids` setting later.

E.g.

```Downloaded Title (GID) -> filename``